### PR TITLE
Add a --qgainmap flag.

### PR DESCRIPTION
**Pull request by maryla-uc**
To control the encoding quality of the gain map.

### CODE REVIEW

File: apps/avifenc.c [hunk 1]
```diff
@@ -230,6 +239,10 @@ static void syntaxLong(void)
     printf("    --ignore-profile,--ignore-icc     : If the input file contains an embedded color profile, ignore it (no-op if absent)\n");
 #if defined(AVIF_ENABLE_EXPERIMENTAL_JPEG_GAIN_MAP_CONVERSION)
     printf("    --ignore-gain-map                 : If the input file contains an embedded gain map, ignore it (no-op if absent)\n");
+    printf("    --qgainmap Q                      : Set quality for the gain map (%d-%d, where %d is lossless)\n",
```
Comment (vrabaud) [inline apps/avifenc.c:242]: Guard with #if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)

File: apps/avifenc.c [hunk 2]
```diff
@@ -32,10 +32,12 @@ typedef struct
     avifCodecChoice codecChoice;
     int jobs;
     int targetSize;
-    avifBool qualityIsConstrained;      // true if quality explicitly set by the user
-    avifBool qualityAlphaIsConstrained; // true if qualityAlpha explicitly set by the user
+    avifBool qualityIsConstrained;        // true if quality explicitly set by the user
+    avifBool qualityAlphaIsConstrained;   // true if qualityAlpha explicitly set by the user
+    avifBool qualityGainMapIsConstrained; // true if qualityGainMap explicitly set by the user
```
Comment (vrabaud) [inline apps/avifenc.c:37]: Do we want to guard that with #if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP) ?

File: apps/avifenc.c [hunk 3]
```diff
@@ -1813,6 +1858,17 @@ int main(int argc, char * argv[])
 #if defined(AVIF_ENABLE_EXPERIMENTAL_JPEG_GAIN_MAP_CONVERSION)
         } else if (!strcmp(arg, "--ignore-gain-map")) {
             settings.ignoreGainMap = AVIF_TRUE;
+        } else if (!strcmp(arg, "--qgainmap")) {
+            NEXTARG();
+            int qualityGainMap = atoi(arg);
+            if (qualityGainMap < AVIF_QUALITY_WORST) {
```
Comment (vrabaud) [inline apps/avifenc.c:1864]: Please use AVIF_CLAMP

File: apps/avifenc.c (hunk 1 above)
Comment (maryla-uc) [reply apps/avifenc.c:245]: AVIF_ENABLE_EXPERIMENTAL_JPEG_GAIN_MAP_CONVERSION implies AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP.
I changed it to use AVIF_ENABLE_EXPERIMENTAL_JPEG_GAIN_MAP_CONVERSION everywhere in this file for simplicity/consistency and added some comments.

File: apps/avifenc.c (hunk 2 above)
Comment (maryla-uc) [reply apps/avifenc.c]: It's debatable, but I think having if-guards everywhere makes the code harder to read, and this variable shouldn't cause any problems if gain maps are off. You can't use the gain map api when gain maps are off anyway.

File: apps/avifenc.c (hunk 3 above)
Comment (maryla-uc) [reply apps/avifenc.c:1867]: It's defined in internal.h which is not available in apps/

File: apps/avifenc.c (hunk 2 above)
Comment (vrabaud) [reply apps/avifenc.c]: Agreed. Plus, this feature will be official one day.

File: apps/avifenc.c (hunk 3 above)
Comment (vrabaud) [reply apps/avifenc.c:1867]: Ah, I thought it was internal to the project, not the lib...

**Review by vrabaud — APPROVED** [review]

**Pull request merged**